# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp
# Commit: b84880c5de44
# Full Hash: b84880c5de443e6274953d210444c57319ce4b5f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 09:18:36
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r c5f28ed5382b -r b84880c5de44 dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp
--- a/dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp	Sat Oct 12 23:42:06 2024 +0300
+++ b/dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp	Sat Oct 12 20:58:36 2024 +0000
@@ -221,18 +221,19 @@
 
 // static
 void QuotaManagerDependencyFixture::InitializeTemporaryOrigin(
-    const OriginMetadata& aOriginMetadata) {
+    const OriginMetadata& aOriginMetadata, bool aCreateIfNonExistent) {
   mozilla::ipc::PrincipalInfo principalInfo;
   ASSERT_NO_FATAL_FAILURE(
       CreateContentPrincipalInfo(aOriginMetadata.mOrigin, principalInfo));
 
   PerformOnBackgroundThread([persistenceType = aOriginMetadata.mPersistenceType,
-                             principalInfo = std::move(principalInfo)]() {
+                             principalInfo = std::move(principalInfo),
+                             aCreateIfNonExistent]() {
     QuotaManager* quotaManager = QuotaManager::Get();
     ASSERT_TRUE(quotaManager);
 
-    Await(quotaManager->InitializeTemporaryOrigin(persistenceType,
-                                                  principalInfo));
+    Await(quotaManager->InitializeTemporaryOrigin(
+        persistenceType, principalInfo, aCreateIfNonExistent));
   });
 }
 